ols/utils/token_handler.py

```diff
@@ -98,47 +101,53 @@ def get_available_tokens(self, prompt: str, model_config: ModelConfig) -> int:
 
     def truncate_rag_context(
         self, retrieved_nodes: list[NodeWithScore], max_tokens: int = 500
-    ) -> tuple[list[dict], int]:
+    ) -> tuple[dict[str, list[str]], int]:
         """Process retrieved node text and truncate if required.
 
         Args:
             retrieved_nodes: retrieved nodes object from index
             max_tokens: maximum tokens allowed for rag context
 
         Returns:
-            context: A list of dictionary containing text & metadata
+            context_dict: A dictionary containing list of context & metadata
             max_tokens: int, available tokens after context usage
             Context Example:
-                [
-                    {
-                        "text": "This is my doc",
-                        "file_name": "doc1.pdf",
-                        "doc_link": link1,
-                    },
-                ]
+                {
+                    "text": ["This is my doc1", "This is my doc2"],
+                    "doc_link": [doc_link2, doc_link2]
```

Comment:
```suggestion
                    "doc_link": [doc_link1, doc_link2]
```

so these are synced parallel arrays?  (text[2] always aligns with doc_link[2] ?)